src/data/polynomial/monic.lean

```diff
@@ -26,6 +26,17 @@ variables {R : Type u} {S : Type v} {a b : R} {m n : ℕ} {ι : Type y}
 section semiring
 variables [semiring R] {p q r : R[X]}
 
+lemma not_monic_zero_iff : ¬ monic (0 : R[X]) ↔ (0 : R) ≠ 1 :=
+by simp [monic]
+
+lemma monic_zero_iff_subsingleton : monic (0 : R[X]) ↔ (∀ a b : R, a = b) :=
```

Comment:
Is there a reason why these lemmas don't use `subsingleton`. That would hook into TC etc... seems like a useful idiom.